src/client/address_book/impl.h

```diff
@@ -63,6 +63,89 @@ class AddressBookSubscriber;
 /// @brief Address book implementation
 class AddressBook : public AddressBookDefaults {
  public:
+  /// @brief Container for Address Book entry
+  class Entry
+  {
+   public:
+    /// @brief No default copy-constructor
+    Entry(Entry& rhs) = delete;
+
+    /// @brief No default move-constructor
+    Entry(Entry&& rhs) = delete;
+
+    /// @brief Create address book entry from a hostname & identity hash
+    /// @param host Kovri hostname
+    /// @param address Kovri identity hash
+    Entry(std::string host, const kovri::core::IdentHash& address)
+        : m_Host(std::move(host)), m_Address(address)
+    {
+      if (m_Host.empty())
+        throw std::runtime_error("AddressBook: empty entry hostname");
+    }
+
+    /// @brief Create address book entry from a host & base64-encoded address
+    /// @param host Kovri hostname
+    /// @param address Base64-encoded Kovri address
+    Entry(std::string host, const std::string& address)
+        : m_Host(std::move(host))
+    {
+      try
+        {
+          if (m_Host.empty())
+            throw std::runtime_error("AddressBook: empty entry hostname");
+          kovri::core::IdentityEx ident;
+          ident.FromBase64(address);
+          m_Address = ident.GetIdentHash();
+        }
+      catch (...)
+        {
+          kovri::core::Exception ex;
+          ex.Dispatch(__func__);
+          throw;
+        }
+    }
+
+    /// @brief Create address book entry from a subscription line
+    /// @param subscription_line Line from a subscription file (host=address format)
+    /// @notes Throws on malformed subscription lines
+    Entry(const std::string& subscription_line)
```

Comment:
Single-parameter constructors should be marked explicit.

Reply:
If I keep this as a constructor, I'll mark it as explicit.